app/ante.go

```diff
@@ -2,54 +2,96 @@ package app
 
 import (
 	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
-	wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
+	wasmTypes "github.com/CosmWasm/wasmd/x/wasm/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
 	"github.com/cosmos/cosmos-sdk/x/auth/ante"
-	"github.com/cosmos/cosmos-sdk/x/auth/signing"
-	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
-	channelkeeper "github.com/cosmos/cosmos-sdk/x/ibc/core/04-channel/keeper"
-	ibcante "github.com/cosmos/cosmos-sdk/x/ibc/core/ante"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	channelkeeper "github.com/cosmos/ibc-go/v2/modules/core/04-channel/keeper"
+	ibcante "github.com/cosmos/ibc-go/v2/modules/core/ante"
 
 	"github.com/confio/tgrade/x/globalfee"
 	"github.com/confio/tgrade/x/poe"
 	poekeeper "github.com/confio/tgrade/x/poe/keeper"
 )
 
-// NewAnteHandler returns an AnteHandler that checks and increments sequence
-// numbers, checks signatures & account numbers, and deducts fees from the first
-// signer.
-func NewAnteHandler(
-	ak ante.AccountKeeper,
-	bankKeeper bankkeeper.SendKeeper,
-	sigGasConsumer ante.SignatureVerificationGasConsumer,
-	signModeHandler signing.SignModeHandler,
-	txCounterStoreKey sdk.StoreKey,
-	channelKeeper channelkeeper.Keeper,
-	paramStore paramtypes.Subspace,
-	contractSource poekeeper.ContractSource,
-	wasmConfig wasmtypes.WasmConfig,
-) sdk.AnteHandler {
-	// initial version copied from sdk https://github.com/cosmos/cosmos-sdk/blob/v0.42.9/x/auth/ante/ante.go
+type bankKeeper interface {
```

Comment:
This is part of the upgrade? Or more related to the vesting contract work?

I'm find to include it but find it a bit odd to define this type here. Maybe import it from x/wasm if it is needed for that interface

Reply:
👌 I have moved this to expected_keepers in poe/types. This is more sdk convention than defining interfaces where they are used.